### PR TITLE
BigQuery snippets

### PR DESCRIPTION
**Pull request by jskeet**
There's almost no new *code* here - it's almost all "see also" entries.

### CODE REVIEW

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryDatasetSnippets.cs [hunk 1]
```diff
@@ -0,0 +1,190 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryDatasetSnippets.cs]: 2017?

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryDatasetSnippets.cs [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            BigQueryClient.Create(projectId).CreateDataset(datasetId);
+
+            // Snippet: Patch(Dataset, bool, *)
+            BigQueryClient client = BigQueryClient.Create(projectId);
+            BigQueryDataset dataset = client.GetDataset(datasetId);
+
+            // There's no ETag in this Dataset. The matchETag parameter in the method call
+            // determines whether the ETag in the original resource is propagated into the
+            // patch.
+            var patch = new Dataset
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryDatasetSnippets.cs]: I know it's obvious what the type is, but I'd rather there is no use of `var` in snippets.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryTableSnippets.cs [hunk 3]
```diff
@@ -84,5 +84,152 @@ public void ListRows()
...
+            BigQueryClient.Create(projectId).CreateTable(datasetId, tableId, new TableSchema());
+
+            // Snippet: Patch(Table, bool, *)
+            BigQueryClient client = BigQueryClient.Create(projectId);
+            BigQueryTable dataset = client.GetTable(datasetId, tableId);
+
+            // There's no ETag in this Table. The matchETag parameter in the method call
+            // determines whether the ETag in the original resource is propagated into the
+            // patch.
+            var patch = new Table
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryTableSnippets.cs]: Don't use `var`

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryDatasetSnippets.cs (hunk 2 above)
Comment (jskeet) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryDatasetSnippets.cs]: Agreed. These were copied and pasted, so I'll check the original source.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryDatasetSnippets.cs (hunk 1 above)
Comment (jskeet) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryDatasetSnippets.cs]: Oh yes, apparently it is...

**Comment by jskeet** [comment]
Thanks, force-pushed the fixes - will merge on green.

**Pull request merged**